In [25]:
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder
import catboost
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import cross_val_score
from sklearn.decomposition import PCA
from imblearn.over_sampling import SMOTE
from sklearn.svm import SVC 
from sklearn.metrics import accuracy_score
from sklearn.pipeline import Pipeline
from sklearn.feature_selection import RFE
from lightgbm import LGBMClassifier
train=pd.read_csv(r'C:\Users\antip\Downloads\kaggle\train_final.csv')
test=pd.read_csv(r'C:\Users\antip\Downloads\kaggle\test_final.csv')
submition=pd.read_csv(r'C:\Users\antip\Downloads\kaggle\sample_submition.csv')
train_x=train.drop('Onshore/Offshore',axis=1).copy()
train_y=train['Onshore/Offshore'].copy()
train_test=pd.concat([train_x,test],axis=0)
train

Tectonic regime  Onshore/Offshore  \
0                                          COMPRESSION           ONSHORE   
1            COMPRESSION/EVAPORITE/EXTENSION/INVERSION          OFFSHORE   
2                                  COMPRESSION/EROSION           ONSHORE   
3    COMPRESSION/EVAPORITE/EXTENSION/GRAVITY/INVERSION          OFFSHORE   
4                                  COMPRESSION/EROSION           ONSHORE   
..                                                 ...               ...   
350                                COMPRESSION/EROSION  ONSHORE-OFFSHORE   
351                                COMPRESSION/EROSION  ONSHORE-OFFSHORE   
352               STRIKE-SLIP/TRANSPRESSION/BASEMENT-I  ONSHORE-OFFSHORE   
353               STRIKE-SLIP/TRANSPRESSION/BASEMENT-I  ONSHORE-OFFSHORE   
354                    INVERSION/COMPRESSION/EXTENSION  ONSHORE-OFFSHORE   

    Hydrocarbon type      Reservoir status  Structural setting         Depth  \
0                OIL     MATURE PRODUCTION            FORELAND  15250.000000   
1     GAS-CONDENSATE    PLATEAU PRODUCTION           RIFT/SALT  16360.000000   
2                OIL  DECLINING PRODUCTION            FORELAND   4700.000000   
3                OIL            DEVELOPING  FORELAND/RIFT/SALT   8105.345849   
4                OIL       NEARLY DEPLETED            FORELAND   7600.000000   
..               ...                   ...                 ...           ...   
350              OIL     MATURE PRODUCTION     WRENCH/FORELAND   1400.000000   
351              OIL       NEARLY DEPLETED            FORELAND   6550.000000   
352              OIL       NEARLY DEPLETED              WRENCH   2326.000000   
353              OIL       NEARLY DEPLETED              WRENCH   1726.000000   
354              OIL  DECLINING PRODUCTION      RIFT/INVERSION   4954.000000   

         Period                Lithology        Gross      Netpay   Porosity  \
0     PALEOGENE                SANDSTONE  8000.000000  575.000000   5.000000   
1      JURASSIC                SANDSTONE   980.000000  490.000000  16.000000   
2       PERMIAN                 DOLOMITE    65.000000   21.000000   8.000000   
3    CRETACEOUS                SANDSTONE   116.037254   41.595776  14.984042   
4    CRETACEOUS                SANDSTONE   158.000000   15.000000   9.500000   
..          ...                      ...          ...         ...        ...   
350     NEOGENE                SANDSTONE   500.000000  120.000000  35.000000   
351  CRETACEOUS                 DOLOMITE  1300.000000  200.000000  11.000000   
352     NEOGENE  THINLY-BEDDED SANDSTONE   700.000000  375.000000  28.000000   
353     NEOGENE  THINLY-BEDDED SANDSTONE   660.000000  152.000000  34.000000   
354    TRIASSIC                SANDSTONE   525.000000  150.000000  18.000000   

     Permeability  
0        0.100000  
1       10.000000  
2        0.500000  
3       29.920211  
4       10.000000  
..            ...  
350   1200.000000  
351     10.000000  
352    400.000000  
353   2300.000000  
354    100.000000  

[355 rows x 12 columns]

In [26]:
le=LabelEncoder()
train_y=le.fit_transform(train_y)
train_y

array([1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1,
       1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0,
       1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1,
       0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0,
       1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0,
       0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0,
       1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
       0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1,
       1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0,
       0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1,
       1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0,

In [27]:
train_test=pd.concat([train_test,train_test['Tectonic regime'].str.get_dummies('/').add_prefix('Tectonic_')]
                  ,axis=1).drop('Tectonic regime',axis=1)
train_test=pd.concat([train_test,train_test['Structural setting'].str.get_dummies('/').add_prefix('Structure_')]
                  ,axis=1).drop('Structural setting',axis=1)
train_test=pd.concat([train_test,train_test['Reservoir status'].str.get_dummies().add_prefix('Reservoir_')]
                  ,axis=1).drop('Reservoir status',axis=1)
train_test=pd.concat([train_test,train_test['Period'].str.get_dummies().add_prefix('Period_')]
                  ,axis=1).drop('Period',axis=1)
train_test=pd.concat([train_test,train_test['Lithology'].str.get_dummies().add_prefix('Lithology_')]
                  ,axis=1).drop('Lithology',axis=1)
train_test=pd.concat([train_test,train_test['Hydrocarbon type'].str.get_dummies().add_prefix('Hydrocarbon_')]
                  ,axis=1).drop('Hydrocarbon type',axis=1)

In [20]:
clf=catboost.CatBoostClassifier(verbose=0)
train_2=train_test.iloc[:train_x.shape[0],:]
test_2=train_test.iloc[train_x.shape[0]:,:]
cross_val_score(clf,train_2,train_y).mean()

0.904225352112676

без pca - 0.904225352112676
pca 50 - 0.922870745900194
pca 70 - 0.9209310527243872
pca 40 - 0.9191147945688591
pca 60 - 0.9322694410156938
rfe 60 lightgbm - 0.943519661435373
rfe 50 lightgbm - 0.9397637101040381
rfe 70 lightgbm - 0.9416328689825427

In [24]:
oversample = SMOTE(k_neighbors=3)
train_3, train_y3 = oversample.fit_resample(train_test.iloc[:train_x.shape[0],:], train_y)
pl_pca=Pipeline([('scaler', StandardScaler()), 
          ('pca', PCA(n_components=60)),
          ('catboost', catboost.CatBoostClassifier(verbose=0))])
cross_val_score(pl_pca,train_3,train_y3).mean()

0.9322694410156938

In [34]:
oversample = SMOTE(k_neighbors=3)
train_3, train_y3 = oversample.fit_resample(train_test.iloc[:train_x.shape[0],:], train_y)
pl_rfe=Pipeline([('scaler', StandardScaler()), 
          ('rfe', RFE(LGBMClassifier(),n_features_to_select=70)),
          ('catboost', catboost.CatBoostClassifier(verbose=1))])
cross_val_score(pl_rfe,train_3,train_y3).mean()

Learning rate set to 0.075739
0:	learn: 1.0186404	total: 3.05ms	remaining: 3.05s
1:	learn: 0.9610710	total: 6.17ms	remaining: 3.08s
2:	learn: 0.9120799	total: 8.76ms	remaining: 2.91s
3:	learn: 0.8584264	total: 11.9ms	remaining: 2.96s
4:	learn: 0.8152136	total: 14.5ms	remaining: 2.88s
5:	learn: 0.7809668	total: 17.1ms	remaining: 2.83s
6:	learn: 0.7477251	total: 20.1ms	remaining: 2.85s
7:	learn: 0.7147094	total: 22.7ms	remaining: 2.81s
8:	learn: 0.6860892	total: 25.2ms	remaining: 2.77s
9:	learn: 0.6579007	total: 27.9ms	remaining: 2.76s
10:	learn: 0.6314430	total: 30.5ms	remaining: 2.74s
11:	learn: 0.6080098	total: 33.1ms	remaining: 2.73s
12:	learn: 0.5856456	total: 36.2ms	remaining: 2.75s
13:	learn: 0.5629505	total: 39.3ms	remaining: 2.77s
14:	learn: 0.5420415	total: 42.2ms	remaining: 2.77s
15:	learn: 0.5228491	total: 44.8ms	remaining: 2.76s
16:	learn: 0.5035391	total: 47.5ms	remaining: 2.75s
17:	learn: 0.4868304	total: 50.6ms	remaining: 2.76s
18:	learn: 0.4716985	total: 53.3ms	remaining

210:	learn: 0.0561490	total: 523ms	remaining: 1.96s
211:	learn: 0.0559177	total: 525ms	remaining: 1.95s
212:	learn: 0.0557186	total: 528ms	remaining: 1.95s
213:	learn: 0.0553998	total: 531ms	remaining: 1.95s
214:	learn: 0.0550421	total: 534ms	remaining: 1.95s
215:	learn: 0.0548168	total: 536ms	remaining: 1.95s
216:	learn: 0.0544795	total: 539ms	remaining: 1.94s
217:	learn: 0.0541112	total: 541ms	remaining: 1.94s
218:	learn: 0.0538894	total: 544ms	remaining: 1.94s
219:	learn: 0.0535025	total: 546ms	remaining: 1.94s
220:	learn: 0.0532835	total: 549ms	remaining: 1.93s
221:	learn: 0.0530459	total: 551ms	remaining: 1.93s
222:	learn: 0.0527423	total: 553ms	remaining: 1.93s
223:	learn: 0.0524699	total: 556ms	remaining: 1.92s
224:	learn: 0.0522609	total: 558ms	remaining: 1.92s
225:	learn: 0.0520841	total: 561ms	remaining: 1.92s
226:	learn: 0.0518515	total: 563ms	remaining: 1.92s
227:	learn: 0.0516657	total: 566ms	remaining: 1.91s
228:	learn: 0.0512956	total: 568ms	remaining: 1.91s
229:	learn: 

423:	learn: 0.0243448	total: 1.04s	remaining: 1.42s
424:	learn: 0.0242576	total: 1.05s	remaining: 1.42s
425:	learn: 0.0241906	total: 1.05s	remaining: 1.42s
426:	learn: 0.0241010	total: 1.05s	remaining: 1.41s
427:	learn: 0.0240313	total: 1.06s	remaining: 1.41s
428:	learn: 0.0239810	total: 1.06s	remaining: 1.41s
429:	learn: 0.0238946	total: 1.06s	remaining: 1.41s
430:	learn: 0.0238421	total: 1.07s	remaining: 1.42s
431:	learn: 0.0237826	total: 1.07s	remaining: 1.41s
432:	learn: 0.0237193	total: 1.08s	remaining: 1.41s
433:	learn: 0.0236864	total: 1.08s	remaining: 1.41s
434:	learn: 0.0236140	total: 1.08s	remaining: 1.41s
435:	learn: 0.0235623	total: 1.08s	remaining: 1.4s
436:	learn: 0.0235012	total: 1.09s	remaining: 1.4s
437:	learn: 0.0234739	total: 1.09s	remaining: 1.4s
438:	learn: 0.0233954	total: 1.09s	remaining: 1.4s
439:	learn: 0.0233119	total: 1.09s	remaining: 1.39s
440:	learn: 0.0232286	total: 1.1s	remaining: 1.39s
441:	learn: 0.0231217	total: 1.1s	remaining: 1.39s
442:	learn: 0.0230

634:	learn: 0.0148636	total: 1.57s	remaining: 902ms
635:	learn: 0.0148345	total: 1.57s	remaining: 900ms
636:	learn: 0.0148138	total: 1.57s	remaining: 897ms
637:	learn: 0.0147851	total: 1.58s	remaining: 895ms
638:	learn: 0.0147682	total: 1.58s	remaining: 893ms
639:	learn: 0.0147264	total: 1.58s	remaining: 890ms
640:	learn: 0.0147002	total: 1.58s	remaining: 888ms
641:	learn: 0.0146683	total: 1.59s	remaining: 885ms
642:	learn: 0.0146540	total: 1.59s	remaining: 883ms
643:	learn: 0.0146224	total: 1.59s	remaining: 880ms
644:	learn: 0.0145845	total: 1.59s	remaining: 878ms
645:	learn: 0.0145623	total: 1.6s	remaining: 875ms
646:	learn: 0.0145264	total: 1.6s	remaining: 873ms
647:	learn: 0.0144932	total: 1.6s	remaining: 870ms
648:	learn: 0.0144660	total: 1.6s	remaining: 868ms
649:	learn: 0.0144471	total: 1.61s	remaining: 865ms
650:	learn: 0.0144198	total: 1.61s	remaining: 863ms
651:	learn: 0.0143958	total: 1.61s	remaining: 861ms
652:	learn: 0.0143560	total: 1.61s	remaining: 858ms
653:	learn: 0.01

843:	learn: 0.0104497	total: 2.09s	remaining: 386ms
844:	learn: 0.0104384	total: 2.09s	remaining: 384ms
845:	learn: 0.0104218	total: 2.1s	remaining: 382ms
846:	learn: 0.0104164	total: 2.1s	remaining: 379ms
847:	learn: 0.0104052	total: 2.1s	remaining: 377ms
848:	learn: 0.0103888	total: 2.1s	remaining: 374ms
849:	learn: 0.0103712	total: 2.11s	remaining: 372ms
850:	learn: 0.0103544	total: 2.11s	remaining: 369ms
851:	learn: 0.0103397	total: 2.11s	remaining: 367ms
852:	learn: 0.0103268	total: 2.11s	remaining: 364ms
853:	learn: 0.0103151	total: 2.12s	remaining: 362ms
854:	learn: 0.0103013	total: 2.12s	remaining: 359ms
855:	learn: 0.0102864	total: 2.12s	remaining: 357ms
856:	learn: 0.0102731	total: 2.12s	remaining: 354ms
857:	learn: 0.0102549	total: 2.13s	remaining: 352ms
858:	learn: 0.0102367	total: 2.13s	remaining: 349ms
859:	learn: 0.0102193	total: 2.13s	remaining: 347ms
860:	learn: 0.0102057	total: 2.13s	remaining: 344ms
861:	learn: 0.0101961	total: 2.13s	remaining: 342ms
862:	learn: 0.01

60:	learn: 0.2099786	total: 166ms	remaining: 2.56s
61:	learn: 0.2065542	total: 169ms	remaining: 2.55s
62:	learn: 0.2041876	total: 171ms	remaining: 2.55s
63:	learn: 0.2004303	total: 174ms	remaining: 2.55s
64:	learn: 0.1987160	total: 177ms	remaining: 2.55s
65:	learn: 0.1966294	total: 180ms	remaining: 2.55s
66:	learn: 0.1938330	total: 183ms	remaining: 2.54s
67:	learn: 0.1915117	total: 186ms	remaining: 2.54s
68:	learn: 0.1887256	total: 188ms	remaining: 2.54s
69:	learn: 0.1862535	total: 190ms	remaining: 2.53s
70:	learn: 0.1839382	total: 193ms	remaining: 2.53s
71:	learn: 0.1824183	total: 196ms	remaining: 2.52s
72:	learn: 0.1792303	total: 198ms	remaining: 2.52s
73:	learn: 0.1767709	total: 201ms	remaining: 2.52s
74:	learn: 0.1745890	total: 204ms	remaining: 2.51s
75:	learn: 0.1729131	total: 206ms	remaining: 2.51s
76:	learn: 0.1713163	total: 209ms	remaining: 2.5s
77:	learn: 0.1692600	total: 211ms	remaining: 2.5s
78:	learn: 0.1674015	total: 214ms	remaining: 2.49s
79:	learn: 0.1660797	total: 217ms

256:	learn: 0.0478715	total: 689ms	remaining: 1.99s
257:	learn: 0.0476115	total: 692ms	remaining: 1.99s
258:	learn: 0.0474350	total: 694ms	remaining: 1.99s
259:	learn: 0.0473262	total: 697ms	remaining: 1.98s
260:	learn: 0.0471143	total: 700ms	remaining: 1.98s
261:	learn: 0.0469702	total: 703ms	remaining: 1.98s
262:	learn: 0.0468023	total: 705ms	remaining: 1.98s
263:	learn: 0.0466629	total: 708ms	remaining: 1.97s
264:	learn: 0.0464615	total: 710ms	remaining: 1.97s
265:	learn: 0.0462508	total: 713ms	remaining: 1.97s
266:	learn: 0.0460127	total: 716ms	remaining: 1.96s
267:	learn: 0.0458915	total: 719ms	remaining: 1.96s
268:	learn: 0.0455805	total: 721ms	remaining: 1.96s
269:	learn: 0.0452982	total: 724ms	remaining: 1.96s
270:	learn: 0.0450979	total: 727ms	remaining: 1.96s
271:	learn: 0.0449000	total: 729ms	remaining: 1.95s
272:	learn: 0.0447380	total: 732ms	remaining: 1.95s
273:	learn: 0.0445492	total: 734ms	remaining: 1.95s
274:	learn: 0.0443429	total: 737ms	remaining: 1.94s
275:	learn: 

460:	learn: 0.0225944	total: 1.21s	remaining: 1.42s
461:	learn: 0.0225270	total: 1.22s	remaining: 1.42s
462:	learn: 0.0224767	total: 1.22s	remaining: 1.41s
463:	learn: 0.0224303	total: 1.22s	remaining: 1.41s
464:	learn: 0.0223885	total: 1.22s	remaining: 1.41s
465:	learn: 0.0223460	total: 1.23s	remaining: 1.41s
466:	learn: 0.0223178	total: 1.23s	remaining: 1.4s
467:	learn: 0.0222731	total: 1.23s	remaining: 1.4s
468:	learn: 0.0222213	total: 1.23s	remaining: 1.4s
469:	learn: 0.0221511	total: 1.24s	remaining: 1.4s
470:	learn: 0.0220936	total: 1.24s	remaining: 1.39s
471:	learn: 0.0220359	total: 1.24s	remaining: 1.39s
472:	learn: 0.0219688	total: 1.24s	remaining: 1.39s
473:	learn: 0.0219038	total: 1.25s	remaining: 1.38s
474:	learn: 0.0218475	total: 1.25s	remaining: 1.38s
475:	learn: 0.0217979	total: 1.25s	remaining: 1.38s
476:	learn: 0.0217232	total: 1.25s	remaining: 1.38s
477:	learn: 0.0216890	total: 1.26s	remaining: 1.37s
478:	learn: 0.0216329	total: 1.26s	remaining: 1.37s
479:	learn: 0.02

665:	learn: 0.0143738	total: 1.74s	remaining: 871ms
666:	learn: 0.0143552	total: 1.74s	remaining: 868ms
667:	learn: 0.0143193	total: 1.74s	remaining: 866ms
668:	learn: 0.0143006	total: 1.74s	remaining: 863ms
669:	learn: 0.0142805	total: 1.75s	remaining: 860ms
670:	learn: 0.0142530	total: 1.75s	remaining: 858ms
671:	learn: 0.0142091	total: 1.75s	remaining: 855ms
672:	learn: 0.0141979	total: 1.75s	remaining: 853ms
673:	learn: 0.0141824	total: 1.76s	remaining: 850ms
674:	learn: 0.0141444	total: 1.76s	remaining: 848ms
675:	learn: 0.0141233	total: 1.76s	remaining: 845ms
676:	learn: 0.0141043	total: 1.76s	remaining: 842ms
677:	learn: 0.0140778	total: 1.77s	remaining: 840ms
678:	learn: 0.0140622	total: 1.77s	remaining: 837ms
679:	learn: 0.0140306	total: 1.77s	remaining: 834ms
680:	learn: 0.0140091	total: 1.77s	remaining: 832ms
681:	learn: 0.0139838	total: 1.78s	remaining: 829ms
682:	learn: 0.0139591	total: 1.78s	remaining: 826ms
683:	learn: 0.0139350	total: 1.78s	remaining: 824ms
684:	learn: 

865:	learn: 0.0104680	total: 2.25s	remaining: 349ms
866:	learn: 0.0104548	total: 2.26s	remaining: 346ms
867:	learn: 0.0104400	total: 2.26s	remaining: 344ms
868:	learn: 0.0104219	total: 2.26s	remaining: 341ms
869:	learn: 0.0104057	total: 2.26s	remaining: 338ms
870:	learn: 0.0103935	total: 2.27s	remaining: 336ms
871:	learn: 0.0103802	total: 2.27s	remaining: 333ms
872:	learn: 0.0103644	total: 2.27s	remaining: 331ms
873:	learn: 0.0103528	total: 2.27s	remaining: 328ms
874:	learn: 0.0103435	total: 2.28s	remaining: 325ms
875:	learn: 0.0103256	total: 2.28s	remaining: 323ms
876:	learn: 0.0103082	total: 2.28s	remaining: 320ms
877:	learn: 0.0103006	total: 2.28s	remaining: 317ms
878:	learn: 0.0102863	total: 2.29s	remaining: 315ms
879:	learn: 0.0102742	total: 2.29s	remaining: 312ms
880:	learn: 0.0102609	total: 2.29s	remaining: 310ms
881:	learn: 0.0102514	total: 2.29s	remaining: 307ms
882:	learn: 0.0102394	total: 2.3s	remaining: 304ms
883:	learn: 0.0102314	total: 2.3s	remaining: 302ms
884:	learn: 0.

65:	learn: 0.1921475	total: 169ms	remaining: 2.39s
66:	learn: 0.1898599	total: 171ms	remaining: 2.39s
67:	learn: 0.1885903	total: 174ms	remaining: 2.38s
68:	learn: 0.1867229	total: 177ms	remaining: 2.38s
69:	learn: 0.1844794	total: 180ms	remaining: 2.39s
70:	learn: 0.1813930	total: 182ms	remaining: 2.38s
71:	learn: 0.1790278	total: 185ms	remaining: 2.39s
72:	learn: 0.1773993	total: 187ms	remaining: 2.38s
73:	learn: 0.1754734	total: 190ms	remaining: 2.37s
74:	learn: 0.1734639	total: 193ms	remaining: 2.37s
75:	learn: 0.1718030	total: 195ms	remaining: 2.37s
76:	learn: 0.1699051	total: 197ms	remaining: 2.36s
77:	learn: 0.1674817	total: 200ms	remaining: 2.36s
78:	learn: 0.1650891	total: 202ms	remaining: 2.36s
79:	learn: 0.1629995	total: 205ms	remaining: 2.35s
80:	learn: 0.1613693	total: 207ms	remaining: 2.35s
81:	learn: 0.1595675	total: 209ms	remaining: 2.34s
82:	learn: 0.1580558	total: 212ms	remaining: 2.34s
83:	learn: 0.1566988	total: 214ms	remaining: 2.34s
84:	learn: 0.1544216	total: 217

271:	learn: 0.0436910	total: 695ms	remaining: 1.86s
272:	learn: 0.0436046	total: 697ms	remaining: 1.86s
273:	learn: 0.0433564	total: 700ms	remaining: 1.85s
274:	learn: 0.0432549	total: 702ms	remaining: 1.85s
275:	learn: 0.0429908	total: 705ms	remaining: 1.85s
276:	learn: 0.0426987	total: 707ms	remaining: 1.85s
277:	learn: 0.0425224	total: 710ms	remaining: 1.84s
278:	learn: 0.0423178	total: 712ms	remaining: 1.84s
279:	learn: 0.0421064	total: 715ms	remaining: 1.84s
280:	learn: 0.0420414	total: 717ms	remaining: 1.83s
281:	learn: 0.0419116	total: 720ms	remaining: 1.83s
282:	learn: 0.0417136	total: 722ms	remaining: 1.83s
283:	learn: 0.0415163	total: 724ms	remaining: 1.82s
284:	learn: 0.0413605	total: 727ms	remaining: 1.82s
285:	learn: 0.0412988	total: 729ms	remaining: 1.82s
286:	learn: 0.0411248	total: 731ms	remaining: 1.82s
287:	learn: 0.0409680	total: 734ms	remaining: 1.81s
288:	learn: 0.0409187	total: 736ms	remaining: 1.81s
289:	learn: 0.0407118	total: 738ms	remaining: 1.81s
290:	learn: 

475:	learn: 0.0225647	total: 1.22s	remaining: 1.34s
476:	learn: 0.0225181	total: 1.22s	remaining: 1.33s
477:	learn: 0.0224431	total: 1.22s	remaining: 1.33s
478:	learn: 0.0223648	total: 1.22s	remaining: 1.33s
479:	learn: 0.0223143	total: 1.23s	remaining: 1.33s
480:	learn: 0.0222821	total: 1.23s	remaining: 1.32s
481:	learn: 0.0222615	total: 1.23s	remaining: 1.32s
482:	learn: 0.0221946	total: 1.23s	remaining: 1.32s
483:	learn: 0.0221299	total: 1.24s	remaining: 1.32s
484:	learn: 0.0220361	total: 1.24s	remaining: 1.31s
485:	learn: 0.0219943	total: 1.24s	remaining: 1.31s
486:	learn: 0.0219834	total: 1.24s	remaining: 1.31s
487:	learn: 0.0219346	total: 1.25s	remaining: 1.31s
488:	learn: 0.0219182	total: 1.25s	remaining: 1.3s
489:	learn: 0.0218655	total: 1.25s	remaining: 1.3s
490:	learn: 0.0218156	total: 1.25s	remaining: 1.3s
491:	learn: 0.0217671	total: 1.25s	remaining: 1.29s
492:	learn: 0.0217526	total: 1.26s	remaining: 1.29s
493:	learn: 0.0216927	total: 1.26s	remaining: 1.29s
494:	learn: 0.0

693:	learn: 0.0145831	total: 1.74s	remaining: 766ms
694:	learn: 0.0145534	total: 1.74s	remaining: 764ms
695:	learn: 0.0145157	total: 1.74s	remaining: 762ms
696:	learn: 0.0144898	total: 1.75s	remaining: 759ms
697:	learn: 0.0144778	total: 1.75s	remaining: 757ms
698:	learn: 0.0144468	total: 1.75s	remaining: 754ms
699:	learn: 0.0144152	total: 1.75s	remaining: 752ms
700:	learn: 0.0143737	total: 1.76s	remaining: 749ms
701:	learn: 0.0143386	total: 1.76s	remaining: 747ms
702:	learn: 0.0143257	total: 1.76s	remaining: 744ms
703:	learn: 0.0143197	total: 1.76s	remaining: 741ms
704:	learn: 0.0143002	total: 1.76s	remaining: 739ms
705:	learn: 0.0142805	total: 1.77s	remaining: 736ms
706:	learn: 0.0142587	total: 1.77s	remaining: 734ms
707:	learn: 0.0142228	total: 1.77s	remaining: 731ms
708:	learn: 0.0142051	total: 1.77s	remaining: 729ms
709:	learn: 0.0141879	total: 1.78s	remaining: 726ms
710:	learn: 0.0141696	total: 1.78s	remaining: 724ms
711:	learn: 0.0141378	total: 1.78s	remaining: 721ms
712:	learn: 

906:	learn: 0.0105150	total: 2.26s	remaining: 232ms
907:	learn: 0.0104989	total: 2.26s	remaining: 229ms
908:	learn: 0.0104864	total: 2.26s	remaining: 227ms
909:	learn: 0.0104829	total: 2.27s	remaining: 224ms
910:	learn: 0.0104695	total: 2.27s	remaining: 222ms
911:	learn: 0.0104529	total: 2.27s	remaining: 219ms
912:	learn: 0.0104388	total: 2.27s	remaining: 217ms
913:	learn: 0.0104251	total: 2.27s	remaining: 214ms
914:	learn: 0.0104074	total: 2.28s	remaining: 212ms
915:	learn: 0.0103872	total: 2.28s	remaining: 209ms
916:	learn: 0.0103805	total: 2.28s	remaining: 207ms
917:	learn: 0.0103592	total: 2.29s	remaining: 204ms
918:	learn: 0.0103356	total: 2.29s	remaining: 202ms
919:	learn: 0.0103209	total: 2.29s	remaining: 199ms
920:	learn: 0.0103095	total: 2.29s	remaining: 197ms
921:	learn: 0.0102913	total: 2.29s	remaining: 194ms
922:	learn: 0.0102825	total: 2.3s	remaining: 192ms
923:	learn: 0.0102678	total: 2.3s	remaining: 189ms
924:	learn: 0.0102545	total: 2.3s	remaining: 187ms
925:	learn: 0.0

129:	learn: 0.0921930	total: 350ms	remaining: 2.35s
130:	learn: 0.0914899	total: 354ms	remaining: 2.34s
131:	learn: 0.0907156	total: 356ms	remaining: 2.34s
132:	learn: 0.0900934	total: 359ms	remaining: 2.34s
133:	learn: 0.0892989	total: 362ms	remaining: 2.34s
134:	learn: 0.0883938	total: 364ms	remaining: 2.33s
135:	learn: 0.0874597	total: 367ms	remaining: 2.33s
136:	learn: 0.0870403	total: 370ms	remaining: 2.33s
137:	learn: 0.0860614	total: 372ms	remaining: 2.33s
138:	learn: 0.0852218	total: 375ms	remaining: 2.33s
139:	learn: 0.0842593	total: 378ms	remaining: 2.32s
140:	learn: 0.0837788	total: 380ms	remaining: 2.32s
141:	learn: 0.0830413	total: 383ms	remaining: 2.31s
142:	learn: 0.0825705	total: 386ms	remaining: 2.31s
143:	learn: 0.0817606	total: 388ms	remaining: 2.31s
144:	learn: 0.0811791	total: 391ms	remaining: 2.31s
145:	learn: 0.0803690	total: 394ms	remaining: 2.3s
146:	learn: 0.0798668	total: 396ms	remaining: 2.3s
147:	learn: 0.0795594	total: 399ms	remaining: 2.3s
148:	learn: 0.0

318:	learn: 0.0326550	total: 873ms	remaining: 1.86s
319:	learn: 0.0325152	total: 875ms	remaining: 1.86s
320:	learn: 0.0324119	total: 878ms	remaining: 1.86s
321:	learn: 0.0322826	total: 881ms	remaining: 1.85s
322:	learn: 0.0321629	total: 884ms	remaining: 1.85s
323:	learn: 0.0320851	total: 887ms	remaining: 1.85s
324:	learn: 0.0320154	total: 889ms	remaining: 1.85s
325:	learn: 0.0318395	total: 892ms	remaining: 1.84s
326:	learn: 0.0317606	total: 895ms	remaining: 1.84s
327:	learn: 0.0316639	total: 897ms	remaining: 1.84s
328:	learn: 0.0315988	total: 900ms	remaining: 1.83s
329:	learn: 0.0314414	total: 903ms	remaining: 1.83s
330:	learn: 0.0313983	total: 905ms	remaining: 1.83s
331:	learn: 0.0312610	total: 908ms	remaining: 1.83s
332:	learn: 0.0312194	total: 911ms	remaining: 1.82s
333:	learn: 0.0311459	total: 913ms	remaining: 1.82s
334:	learn: 0.0310269	total: 916ms	remaining: 1.82s
335:	learn: 0.0308463	total: 919ms	remaining: 1.82s
336:	learn: 0.0307263	total: 922ms	remaining: 1.81s
337:	learn: 

513:	learn: 0.0181678	total: 1.4s	remaining: 1.32s
514:	learn: 0.0181158	total: 1.4s	remaining: 1.32s
515:	learn: 0.0180723	total: 1.4s	remaining: 1.31s
516:	learn: 0.0180125	total: 1.41s	remaining: 1.31s
517:	learn: 0.0179933	total: 1.41s	remaining: 1.31s
518:	learn: 0.0179583	total: 1.41s	remaining: 1.31s
519:	learn: 0.0179238	total: 1.41s	remaining: 1.3s
520:	learn: 0.0178981	total: 1.42s	remaining: 1.3s
521:	learn: 0.0178840	total: 1.42s	remaining: 1.3s
522:	learn: 0.0178380	total: 1.42s	remaining: 1.3s
523:	learn: 0.0178062	total: 1.43s	remaining: 1.29s
524:	learn: 0.0177510	total: 1.43s	remaining: 1.29s
525:	learn: 0.0177026	total: 1.43s	remaining: 1.29s
526:	learn: 0.0176322	total: 1.43s	remaining: 1.29s
527:	learn: 0.0176043	total: 1.44s	remaining: 1.28s
528:	learn: 0.0175489	total: 1.44s	remaining: 1.28s
529:	learn: 0.0175027	total: 1.44s	remaining: 1.28s
530:	learn: 0.0174689	total: 1.45s	remaining: 1.28s
531:	learn: 0.0174029	total: 1.45s	remaining: 1.27s
532:	learn: 0.01736

703:	learn: 0.0123948	total: 1.92s	remaining: 808ms
704:	learn: 0.0123844	total: 1.93s	remaining: 806ms
705:	learn: 0.0123695	total: 1.93s	remaining: 803ms
706:	learn: 0.0123481	total: 1.93s	remaining: 801ms
707:	learn: 0.0123284	total: 1.93s	remaining: 798ms
708:	learn: 0.0123058	total: 1.94s	remaining: 796ms
709:	learn: 0.0122858	total: 1.94s	remaining: 793ms
710:	learn: 0.0122662	total: 1.94s	remaining: 790ms
711:	learn: 0.0122483	total: 1.95s	remaining: 788ms
712:	learn: 0.0122152	total: 1.95s	remaining: 785ms
713:	learn: 0.0121910	total: 1.95s	remaining: 782ms
714:	learn: 0.0121762	total: 1.96s	remaining: 780ms
715:	learn: 0.0121609	total: 1.96s	remaining: 777ms
716:	learn: 0.0121321	total: 1.96s	remaining: 774ms
717:	learn: 0.0121079	total: 1.96s	remaining: 772ms
718:	learn: 0.0120893	total: 1.97s	remaining: 769ms
719:	learn: 0.0120711	total: 1.97s	remaining: 766ms
720:	learn: 0.0120454	total: 1.97s	remaining: 763ms
721:	learn: 0.0120228	total: 1.98s	remaining: 761ms
722:	learn: 

888:	learn: 0.0093221	total: 2.44s	remaining: 305ms
889:	learn: 0.0093130	total: 2.45s	remaining: 302ms
890:	learn: 0.0093027	total: 2.45s	remaining: 300ms
891:	learn: 0.0092959	total: 2.45s	remaining: 297ms
892:	learn: 0.0092881	total: 2.46s	remaining: 294ms
893:	learn: 0.0092710	total: 2.46s	remaining: 291ms
894:	learn: 0.0092535	total: 2.46s	remaining: 289ms
895:	learn: 0.0092408	total: 2.46s	remaining: 286ms
896:	learn: 0.0092254	total: 2.47s	remaining: 283ms
897:	learn: 0.0092118	total: 2.47s	remaining: 281ms
898:	learn: 0.0092001	total: 2.47s	remaining: 278ms
899:	learn: 0.0091805	total: 2.48s	remaining: 275ms
900:	learn: 0.0091673	total: 2.48s	remaining: 272ms
901:	learn: 0.0091529	total: 2.48s	remaining: 269ms
902:	learn: 0.0091388	total: 2.48s	remaining: 267ms
903:	learn: 0.0091215	total: 2.48s	remaining: 264ms
904:	learn: 0.0091146	total: 2.49s	remaining: 261ms
905:	learn: 0.0091041	total: 2.49s	remaining: 258ms
906:	learn: 0.0090869	total: 2.49s	remaining: 256ms
907:	learn: 

65:	learn: 0.1873321	total: 174ms	remaining: 2.47s
66:	learn: 0.1848862	total: 177ms	remaining: 2.46s
67:	learn: 0.1830977	total: 179ms	remaining: 2.46s
68:	learn: 0.1809296	total: 185ms	remaining: 2.49s
69:	learn: 0.1787400	total: 187ms	remaining: 2.49s
70:	learn: 0.1759844	total: 191ms	remaining: 2.49s
71:	learn: 0.1734270	total: 193ms	remaining: 2.49s
72:	learn: 0.1717912	total: 195ms	remaining: 2.48s
73:	learn: 0.1706947	total: 198ms	remaining: 2.48s
74:	learn: 0.1684159	total: 201ms	remaining: 2.47s
75:	learn: 0.1664112	total: 203ms	remaining: 2.47s
76:	learn: 0.1642247	total: 206ms	remaining: 2.47s
77:	learn: 0.1627018	total: 208ms	remaining: 2.46s
78:	learn: 0.1607828	total: 211ms	remaining: 2.46s
79:	learn: 0.1590840	total: 214ms	remaining: 2.46s
80:	learn: 0.1575181	total: 216ms	remaining: 2.45s
81:	learn: 0.1558403	total: 219ms	remaining: 2.45s
82:	learn: 0.1537611	total: 222ms	remaining: 2.45s
83:	learn: 0.1516688	total: 224ms	remaining: 2.44s
84:	learn: 0.1501007	total: 226

269:	learn: 0.0396240	total: 701ms	remaining: 1.89s
270:	learn: 0.0394666	total: 704ms	remaining: 1.89s
271:	learn: 0.0392674	total: 707ms	remaining: 1.89s
272:	learn: 0.0391341	total: 710ms	remaining: 1.89s
273:	learn: 0.0389588	total: 713ms	remaining: 1.89s
274:	learn: 0.0387641	total: 715ms	remaining: 1.89s
275:	learn: 0.0386100	total: 718ms	remaining: 1.88s
276:	learn: 0.0384443	total: 721ms	remaining: 1.88s
277:	learn: 0.0382784	total: 723ms	remaining: 1.88s
278:	learn: 0.0381694	total: 725ms	remaining: 1.87s
279:	learn: 0.0379763	total: 728ms	remaining: 1.87s
280:	learn: 0.0378448	total: 730ms	remaining: 1.87s
281:	learn: 0.0376264	total: 733ms	remaining: 1.87s
282:	learn: 0.0375540	total: 735ms	remaining: 1.86s
283:	learn: 0.0374373	total: 738ms	remaining: 1.86s
284:	learn: 0.0372755	total: 741ms	remaining: 1.86s
285:	learn: 0.0370736	total: 743ms	remaining: 1.85s
286:	learn: 0.0368553	total: 746ms	remaining: 1.85s
287:	learn: 0.0366501	total: 748ms	remaining: 1.85s
288:	learn: 

471:	learn: 0.0203051	total: 1.22s	remaining: 1.37s
472:	learn: 0.0202501	total: 1.22s	remaining: 1.36s
473:	learn: 0.0201666	total: 1.23s	remaining: 1.36s
474:	learn: 0.0201314	total: 1.23s	remaining: 1.36s
475:	learn: 0.0200897	total: 1.23s	remaining: 1.36s
476:	learn: 0.0200434	total: 1.24s	remaining: 1.35s
477:	learn: 0.0200077	total: 1.24s	remaining: 1.35s
478:	learn: 0.0199219	total: 1.24s	remaining: 1.35s
479:	learn: 0.0198851	total: 1.24s	remaining: 1.34s
480:	learn: 0.0198477	total: 1.24s	remaining: 1.34s
481:	learn: 0.0197970	total: 1.25s	remaining: 1.34s
482:	learn: 0.0197471	total: 1.25s	remaining: 1.34s
483:	learn: 0.0196790	total: 1.25s	remaining: 1.33s
484:	learn: 0.0196108	total: 1.25s	remaining: 1.33s
485:	learn: 0.0195398	total: 1.26s	remaining: 1.33s
486:	learn: 0.0194932	total: 1.26s	remaining: 1.33s
487:	learn: 0.0194428	total: 1.26s	remaining: 1.32s
488:	learn: 0.0193970	total: 1.26s	remaining: 1.32s
489:	learn: 0.0193622	total: 1.27s	remaining: 1.32s
490:	learn: 

678:	learn: 0.0128922	total: 1.74s	remaining: 825ms
679:	learn: 0.0128695	total: 1.75s	remaining: 823ms
680:	learn: 0.0128464	total: 1.75s	remaining: 820ms
681:	learn: 0.0128188	total: 1.75s	remaining: 817ms
682:	learn: 0.0127964	total: 1.75s	remaining: 815ms
683:	learn: 0.0127615	total: 1.76s	remaining: 812ms
684:	learn: 0.0127354	total: 1.76s	remaining: 810ms
685:	learn: 0.0127098	total: 1.76s	remaining: 808ms
686:	learn: 0.0126950	total: 1.77s	remaining: 805ms
687:	learn: 0.0126646	total: 1.77s	remaining: 802ms
688:	learn: 0.0126476	total: 1.77s	remaining: 800ms
689:	learn: 0.0126301	total: 1.77s	remaining: 797ms
690:	learn: 0.0125957	total: 1.78s	remaining: 795ms
691:	learn: 0.0125697	total: 1.78s	remaining: 792ms
692:	learn: 0.0125549	total: 1.78s	remaining: 790ms
693:	learn: 0.0125105	total: 1.78s	remaining: 787ms
694:	learn: 0.0124818	total: 1.79s	remaining: 785ms
695:	learn: 0.0124474	total: 1.79s	remaining: 782ms
696:	learn: 0.0124197	total: 1.79s	remaining: 780ms
697:	learn: 

872:	learn: 0.0094344	total: 2.27s	remaining: 330ms
873:	learn: 0.0094257	total: 2.27s	remaining: 328ms
874:	learn: 0.0094177	total: 2.28s	remaining: 325ms
875:	learn: 0.0094055	total: 2.28s	remaining: 323ms
876:	learn: 0.0093965	total: 2.28s	remaining: 320ms
877:	learn: 0.0093848	total: 2.29s	remaining: 318ms
878:	learn: 0.0093727	total: 2.29s	remaining: 315ms
879:	learn: 0.0093583	total: 2.29s	remaining: 312ms
880:	learn: 0.0093490	total: 2.29s	remaining: 310ms
881:	learn: 0.0093390	total: 2.29s	remaining: 307ms
882:	learn: 0.0093234	total: 2.3s	remaining: 305ms
883:	learn: 0.0093107	total: 2.3s	remaining: 302ms
884:	learn: 0.0093046	total: 2.3s	remaining: 299ms
885:	learn: 0.0092935	total: 2.31s	remaining: 297ms
886:	learn: 0.0092786	total: 2.31s	remaining: 294ms
887:	learn: 0.0092695	total: 2.31s	remaining: 291ms
888:	learn: 0.0092613	total: 2.31s	remaining: 289ms
889:	learn: 0.0092442	total: 2.32s	remaining: 286ms
890:	learn: 0.0092339	total: 2.32s	remaining: 284ms
891:	learn: 0.0

0.9416328689825427

In [35]:
pl_rfe=Pipeline([('scaler', StandardScaler()), 
          ('rfe', RFE(LGBMClassifier(),n_features_to_select=60)),
          ('catboost', catboost.CatBoostClassifier(verbose=1))])
pl_rfe.fit(train_3, train_y3)

Learning rate set to 0.076613
0:	learn: 1.0229551	total: 2.81ms	remaining: 2.8s
1:	learn: 0.9533475	total: 5.59ms	remaining: 2.79s
2:	learn: 0.8932942	total: 8.74ms	remaining: 2.9s
3:	learn: 0.8420889	total: 11.6ms	remaining: 2.88s
4:	learn: 0.7989303	total: 14.5ms	remaining: 2.89s
5:	learn: 0.7635925	total: 17.2ms	remaining: 2.84s
6:	learn: 0.7245476	total: 19.8ms	remaining: 2.81s
7:	learn: 0.6892502	total: 22.3ms	remaining: 2.76s
8:	learn: 0.6569108	total: 24.9ms	remaining: 2.74s
9:	learn: 0.6266097	total: 27.5ms	remaining: 2.72s
10:	learn: 0.6018419	total: 30ms	remaining: 2.7s
11:	learn: 0.5767448	total: 32.5ms	remaining: 2.68s
12:	learn: 0.5553239	total: 35.4ms	remaining: 2.69s
13:	learn: 0.5345950	total: 38ms	remaining: 2.67s
14:	learn: 0.5161187	total: 40.7ms	remaining: 2.67s
15:	learn: 0.5007581	total: 43.4ms	remaining: 2.67s
16:	learn: 0.4840391	total: 46.1ms	remaining: 2.67s
17:	learn: 0.4660345	total: 48.7ms	remaining: 2.66s
18:	learn: 0.4512692	total: 51.3ms	remaining: 2.65s

215:	learn: 0.0528685	total: 523ms	remaining: 1.9s
216:	learn: 0.0525748	total: 526ms	remaining: 1.9s
217:	learn: 0.0521965	total: 528ms	remaining: 1.89s
218:	learn: 0.0519455	total: 531ms	remaining: 1.89s
219:	learn: 0.0517597	total: 533ms	remaining: 1.89s
220:	learn: 0.0516224	total: 536ms	remaining: 1.89s
221:	learn: 0.0513836	total: 539ms	remaining: 1.89s
222:	learn: 0.0511167	total: 541ms	remaining: 1.89s
223:	learn: 0.0508553	total: 544ms	remaining: 1.88s
224:	learn: 0.0506243	total: 546ms	remaining: 1.88s
225:	learn: 0.0504446	total: 549ms	remaining: 1.88s
226:	learn: 0.0502290	total: 551ms	remaining: 1.88s
227:	learn: 0.0499730	total: 553ms	remaining: 1.87s
228:	learn: 0.0496645	total: 556ms	remaining: 1.87s
229:	learn: 0.0494291	total: 558ms	remaining: 1.87s
230:	learn: 0.0491961	total: 560ms	remaining: 1.86s
231:	learn: 0.0489874	total: 563ms	remaining: 1.86s
232:	learn: 0.0487160	total: 565ms	remaining: 1.86s
233:	learn: 0.0485074	total: 567ms	remaining: 1.86s
234:	learn: 0.

436:	learn: 0.0232135	total: 1.05s	remaining: 1.35s
437:	learn: 0.0231211	total: 1.05s	remaining: 1.35s
438:	learn: 0.0230365	total: 1.05s	remaining: 1.34s
439:	learn: 0.0229395	total: 1.05s	remaining: 1.34s
440:	learn: 0.0228728	total: 1.06s	remaining: 1.34s
441:	learn: 0.0228392	total: 1.06s	remaining: 1.34s
442:	learn: 0.0227713	total: 1.06s	remaining: 1.33s
443:	learn: 0.0226959	total: 1.06s	remaining: 1.33s
444:	learn: 0.0226399	total: 1.07s	remaining: 1.33s
445:	learn: 0.0225690	total: 1.07s	remaining: 1.33s
446:	learn: 0.0224885	total: 1.07s	remaining: 1.32s
447:	learn: 0.0224152	total: 1.07s	remaining: 1.32s
448:	learn: 0.0223796	total: 1.08s	remaining: 1.32s
449:	learn: 0.0223223	total: 1.08s	remaining: 1.32s
450:	learn: 0.0222701	total: 1.08s	remaining: 1.31s
451:	learn: 0.0222143	total: 1.08s	remaining: 1.31s
452:	learn: 0.0221754	total: 1.08s	remaining: 1.31s
453:	learn: 0.0221014	total: 1.09s	remaining: 1.31s
454:	learn: 0.0220453	total: 1.09s	remaining: 1.3s
455:	learn: 0

655:	learn: 0.0140394	total: 1.57s	remaining: 825ms
656:	learn: 0.0140021	total: 1.57s	remaining: 822ms
657:	learn: 0.0139720	total: 1.58s	remaining: 820ms
658:	learn: 0.0139425	total: 1.58s	remaining: 817ms
659:	learn: 0.0139300	total: 1.58s	remaining: 815ms
660:	learn: 0.0139049	total: 1.58s	remaining: 813ms
661:	learn: 0.0138739	total: 1.59s	remaining: 811ms
662:	learn: 0.0138512	total: 1.59s	remaining: 808ms
663:	learn: 0.0138288	total: 1.59s	remaining: 806ms
664:	learn: 0.0137981	total: 1.59s	remaining: 803ms
665:	learn: 0.0137530	total: 1.6s	remaining: 801ms
666:	learn: 0.0137095	total: 1.6s	remaining: 799ms
667:	learn: 0.0137019	total: 1.6s	remaining: 796ms
668:	learn: 0.0136734	total: 1.6s	remaining: 794ms
669:	learn: 0.0136500	total: 1.61s	remaining: 791ms
670:	learn: 0.0136184	total: 1.61s	remaining: 789ms
671:	learn: 0.0135890	total: 1.61s	remaining: 786ms
672:	learn: 0.0135764	total: 1.61s	remaining: 784ms
673:	learn: 0.0135619	total: 1.61s	remaining: 782ms
674:	learn: 0.01

872:	learn: 0.0097735	total: 2.1s	remaining: 305ms
873:	learn: 0.0097555	total: 2.1s	remaining: 303ms
874:	learn: 0.0097419	total: 2.1s	remaining: 300ms
875:	learn: 0.0097269	total: 2.1s	remaining: 298ms
876:	learn: 0.0097115	total: 2.11s	remaining: 296ms
877:	learn: 0.0096975	total: 2.11s	remaining: 293ms
878:	learn: 0.0096835	total: 2.11s	remaining: 291ms
879:	learn: 0.0096753	total: 2.11s	remaining: 288ms
880:	learn: 0.0096599	total: 2.12s	remaining: 286ms
881:	learn: 0.0096492	total: 2.12s	remaining: 284ms
882:	learn: 0.0096382	total: 2.12s	remaining: 281ms
883:	learn: 0.0096212	total: 2.12s	remaining: 279ms
884:	learn: 0.0096016	total: 2.13s	remaining: 276ms
885:	learn: 0.0095843	total: 2.13s	remaining: 274ms
886:	learn: 0.0095680	total: 2.13s	remaining: 272ms
887:	learn: 0.0095550	total: 2.13s	remaining: 269ms
888:	learn: 0.0095407	total: 2.14s	remaining: 267ms
889:	learn: 0.0095242	total: 2.14s	remaining: 264ms
890:	learn: 0.0095138	total: 2.14s	remaining: 262ms
891:	learn: 0.00

Pipeline(steps=[('scaler', StandardScaler()),
                ('rfe',
                 RFE(estimator=LGBMClassifier(), n_features_to_select=60)),
                ('catboost',
                 <catboost.core.CatBoostClassifier object at 0x000001B878BCAA00>)])

In [56]:
df=pd.DataFrame(le.inverse_transform(pl_rfe.predict(train_test.iloc[train_x.shape[0]:,:])),columns=['Onshore/Offshore']) 
df.index.rename('Index',inplace=True)
df.to_csv(r'C:\Users\antip\Desktop\a.csv')

c:\users\antip\appdata\local\programs\python\python38\lib\site-packages\sklearn\preprocessing\_label.py:154: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
